lib/common/tests/schemas/pcmk__cmp_schemas_by_name_test.c

```diff
@@ -0,0 +1,98 @@
+
+static void
+unknown_is_lesser(void **state)
+{
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-0.1",
+                                          "pacemaker-0.2") == 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-0.1",
+                                          "pacemaker-1.0") < 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.0",
+                                          "pacemaker-0.1") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-0.6",
+                                          "pacemaker-next") < 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-next",
+                                          "pacemaker-0.7") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.1", NULL) < 0);
+    assert_true(pcmk__cmp_schemas_by_name(NULL, "pacemaker-0.0") > 0);
+}
+
+static void
+none_is_greater(void **state)
+{
+    assert_true(pcmk__cmp_schemas_by_name(NULL, NULL) == 0);
+    assert_true(pcmk__cmp_schemas_by_name(NULL, PCMK_VALUE_NONE) == 0);
+    assert_true(pcmk__cmp_schemas_by_name(PCMK_VALUE_NONE, NULL) == 0);
+    assert_true(pcmk__cmp_schemas_by_name(PCMK_VALUE_NONE,
+                                          PCMK_VALUE_NONE) == 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-next",
+                                          PCMK_VALUE_NONE) < 0);
+    assert_true(pcmk__cmp_schemas_by_name(PCMK_VALUE_NONE,
+                                          "pacemaker-next") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-3.0",
+                                          PCMK_VALUE_NONE) < 0);
+    assert_true(pcmk__cmp_schemas_by_name(PCMK_VALUE_NONE,
+                                          "pacemaker-1.0") > 0);
+}
+
+static void
+next_is_before_none(void **state)
+{
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-next",
+                                          "pacemaker-next") == 0);
+    assert_true(pcmk__cmp_schemas_by_name(NULL, "pacemaker-next") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-next", NULL) < 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-3.0",
+                                          "pacemaker-next") < 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-next",
+                                          "pacemaker-1.0") > 0);
+}
+
+static void
+known_numeric(void **state)
+{
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.0",
+                                          "pacemaker-1.0") == 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.2",
+                                          "pacemaker-1.0") > 0);
+    assert_true(pcmk__cmp_schemas_by_name("pacemaker-1.2",
+                                          "pacemaker-2.0") < 0);
+}
+
+PCMK__UNIT_TEST(setup, teardown,
+                cmocka_unit_test(unknown_is_lesser),
+                cmocka_unit_test(none_is_greater),
+                cmocka_unit_test(next_is_before_none),
+                cmocka_unit_test(known_numeric));
```

Comment:
It'd be worth adding a test here and there that uses uppercase in the schema names.  It looks like pcmk__get_schema uses case-insensitive comparisons.

Reply:
added commit on top